src/utils/daos.c

```diff
@@ -1171,18 +1158,33 @@ cont_op_hdlr(struct cmd_args_s *ap)
 			ap->oid = il_reply.fir_oid;
 		} else {
 			ap->type = dattr.da_type;
-			uuid_copy(ap->p_uuid, dattr.da_puuid);
-			uuid_copy(ap->c_uuid, dattr.da_cuuid);
+
+			/** set pool/cont label or uuid */
+			if (dattr.da_pool_label)
+				ap->pool_label = dattr.da_pool_label;
+			else
+				uuid_copy(ap->p_uuid, dattr.da_puuid);
+
+			if (dattr.da_cont_label)
+				ap->cont_label = dattr.da_cont_label;
+			else
+				uuid_copy(ap->c_uuid, dattr.da_cuuid);
+
 			if (dattr.da_rel_path)
 				free(dattr.da_rel_path);
 		}
 	} else {
 		ARGS_VERIFY_PUUID(ap, out, rc = RC_PRINT_HELP);
 	}
 
-	rc = daos_pool_connect(ap->p_uuid, ap->sysname, DAOS_PC_RW,
-			       &ap->pool, NULL /* info */,
-			       NULL /* ev */);
+	if (ap->pool_label)
+		rc = daos_pool_connect_by_label(ap->pool_label, ap->sysname,
+						DAOS_PC_RW, &ap->pool,
+						NULL, NULL);
```

Comment:
In dfuse at least we want to know the uuid even after connecting so we pass in an info pointer here, and then uuid_copy() the data back to where we want it, would it make sense to update p_uuid in this case?

Reply:
yes we could do that, but im not sure we need the uuid after connection. at least for what i tried the uuid was not needed. i'll keep that in mind though if we do for later.

Reply:
Depends on the use-case, certainly for dfuse we do but then it might be unique and it's not using this code.